docs/source/conf.py

```diff
@@ -125,8 +125,10 @@ class MyReferenceStyle(AuthorYearReferenceStyle):
     ("py:class", "dataclasses.InitVar"),
     ("py:class", "Quantity"),
     ("py:class", "numpy._typing._array_like._ScalarType_co"),
-    # TODO - Delete this once Vivienne has merged this feature into develop
-    ("py:class", "virtual_ecosystem.models.abiotic.energy_balance.EnergyBalance"),
+    # God only knows why this is needed. We don't refer to pint.util.Quantity and it
+    # isn't in the pint objects.inv, so why the hell is intersphinx trying to build
+    # references to it.
+    ("py:class", "pint.util.Quantity"),
```

Comment:
Wait, should this be `pint.Quantity`, as you've written it below? I'm confused

Reply:
I'm also confused. There's also a `Quantity` up in line 126, which I'm not sure if it is the same quantity.

Reply:
@alexdewar and @dalonsoa

I should have made this clearer. Asides from updating the `pyproject.toml` references and adding the Ruff rule, there were basically only about 8 lines that needed changing (`np.infty`, `np.float_` and `np.column_stack`) and it looks like all those changes have been made in branches that have already merged to `develop`.

The `pint` stuff is a pain - we needed to update `pint` as our current version wasn't `numpy 2.0` compliant, but there is something off with the `pint` documentation and `objects.inv` that means that whatever we seem to do, we end up with `sphinx` choking on this mysterious `pint.util.Quantity`. There may be a fix - I'm similarly puzzled by why we need so many `numpy` nitpick ignores.